lib/ansible/plugins/inventory/docker_machine.py

```diff
@@ -0,0 +1,271 @@
+
+                id = self.node_attrs['Driver']['MachineName']
+
+                # query `docker-machine env` to obtain remote Docker daemon connection settings in the form of commands
+                # that could be used to set environment variables to influence a local Docker client:
+                env_var_tuples = self._get_docker_daemon_variables(id)
+                if self._should_skip_host(id, env_var_tuples):
+                    continue
+
+                # add an entry in the inventory for this host
+                self.inventory.add_host(id)
+
+                # set standard Ansible remote host connection settings to details captured from `docker-machine`
+                # see: https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html
+                self.inventory.set_variable(id, 'ansible_host', self.node_attrs['Driver']['IPAddress'])
+                self.inventory.set_variable(id, 'ansible_port', self.node_attrs['Driver']['SSHPort'])
+                self.inventory.set_variable(id, 'ansible_user', self.node_attrs['Driver']['SSHUser'])
+                self.inventory.set_variable(id, 'ansible_ssh_private_key_file', self.node_attrs['Driver']['SSHKeyPath'])
+
+                # set variables based on Docker Machine tags
+                self._set_tag_variables(id)
+
+                # set variables based on Docker Machine env variables
+                for kv in env_var_tuples:
+                    self.inventory.set_variable(id, 'dm_{0}'.format(kv[0]), kv[1])
+
+                if self.get_option('verbose_output'):
+                    self.inventory.set_variable(id, 'docker_machine_node_attributes', self.node_attrs)
+
+                # Use constructed if applicable
+                strict = self.get_option('strict')
+
+                # Composed variables
+                self._set_composite_vars(self.get_option('compose'), self.node_attrs, id, strict=strict)
+
+                # Complex groups based on jinja2 conditionals, hosts that meet the conditional are added to group
+                self._add_host_to_composed_groups(self.get_option('groups'), self.node_attrs, id, strict=strict)
+
+                # Create groups based on variable values and add the corresponding hosts to it
+                self._add_host_to_keyed_groups(self.get_option('keyed_groups'), self.node_attrs, id, strict=strict)
+
+        except Exception as e:
+            raise AnsibleError('Unable to fetch hosts from Docker Machine, this was the original exception: %s' %
+                               to_native(e))
+
+    def verify_file(self, path):
+        """Return the possibility of a file being consumable by this plugin."""
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(('docker_machine.yml', 'docker_machine.yaml')):
+                return True
+        display.debug("docker_machine inventory filename must end with 'docker_machine.yml' or 'docker_machine.yaml'")
```

Comment:
not sure this debug is useful

Reply:
HI @bcoca,
That debug was copied from the aws ec2 plugin, and I actually benefited
from it when I couldn't work out why my file with the wrong name was not
being loaded as the standard error message was not helpful, at least for me.

Reply:
Since it's the only plugin which does that, I would remove it. Either all plugins should do that, or none.